tests/testutils.py

```diff
@@ -213,6 +215,29 @@ def assert_run_python_script(source_code, timeout=TIMEOUT):
         os.unlink(source_file)
 
 
+def check_determinist_pickle(a, b):
+    """Check that two pickle output are the same.
+
+    If it is not the case, print the diff between the disassembled pickle
+    payloads.
+
+    This helper is useful to investigate non-deterministic pickling.
+    """
+    if a != b:
+        with io.StringIO() as out:
+            pickletools.dis(pickletools.optimize(a), out)
+            a_out = out.getvalue()
+            # Remove the 11 first characters of each line to remove the bytecode offset
+            # of each object, which is different on each line for very small differences,
+            # making the diff very hard to read.
+            a_out = "\n".join(ll[11:] for ll in a_out.splitlines())
+        with io.StringIO() as out:
+            pickletools.dis(pickletools.optimize(b), out)
+            b_out = out.getvalue()
+            b_out = "\n".join(ll[11:] for ll in b_out.splitlines())
+        assert a_out == b_out
```

Comment:
By importing difflib from the standard library, it should be possible to display a more significan part of full diff instead of just the first line that differs as pytest does by default:



```suggestion
        full_diff = difflib.context_diff(
            a_out.splitlines(keepends=True), b_out.splitlines(keepends=True)
        )
        full_diff = "".join(full_diff)
        if len(full_diff) > 1000:
            full_diff = full_diff[:994] + " [...]"
        raise AssertionError(
           "Pickle payloads are not bitwise equal:\n"
           + full_diff
        )
```